ambry-api/src/main/java/com.github.ambry/account/AccountBuilder.java

```diff
@@ -0,0 +1,216 @@
+/*
+ * Copyright 2017 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.account;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import static com.github.ambry.account.Account.*;
+import static com.github.ambry.account.Container.*;
+
+
+/**
+ * A builder class for {@link Account}. Since {@link Account} is immutable, modifying an {@link Account} needs to
+ * build a new {@link Account} object with updated fields through this builder. An {@link Account} can be built
+ * in two ways: 1) from an existing {@link Account} object; and 2) by supplying required fields of an {@link Account}.
+ */
+public class AccountBuilder {
+  private Short id;
+  private String name;
+  private Account.AccountStatus status;
+  private short version;
+  private Map<Short, Container> idToContainerMetadataMap = new HashMap<>();
+
+  /**
+   * Constructor. This will allow building a new {@link Account} from an existing {@link Account}. The builder will
+   * include all the information including the {@link Container}s of the existing {@link Account}.
+   * @param origin The {@link Account} to build from.
+   */
+  public AccountBuilder(Account origin) {
+    if (origin == null) {
+      throw new IllegalArgumentException("origin cannot be null.");
+    }
+    this.id = origin.getId();
+    this.name = origin.getName();
+    this.status = origin.getStatus();
+    for (Container container : origin.getAllContainers()) {
+      idToContainerMetadataMap.put(container.getId(), container);
+    }
+    this.version = ACCOUNT_METADATA_VERSION_1;
+  }
+
+  /**
+   * Constructor. The builder will not include any {@link Container} information.
+   * @param id The id of the {@link Account} to build.
+   * @param name The name of the {@link Account}. Cannot be {@code null}.
+   * @param status The status of the {@link Account}. Cannot be {@code null}.
+   */
+  public AccountBuilder(short id, String name, AccountStatus status) {
+    this.id = id;
+    this.name = name;
+    this.status = status;
+    this.version = ACCOUNT_METADATA_VERSION_1;
+  }
+
+  /**
+   * Sets the id of the {@link Account} to build.
+   * @param id The id to set.
+   * @return This builder.
+   */
+  public AccountBuilder setId(short id) {
```

Comment:
One thing to think about is that changing the account id would make the parent account id in any child containers invalid. We could either recreate all of the containers currently in `idToContainerMetadataMap`, or we could get rid of this method, since it is a required field from the constructor anyways.